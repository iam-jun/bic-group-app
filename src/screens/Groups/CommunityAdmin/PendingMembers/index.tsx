import {
  View,
  StyleSheet,
  FlatList,
  ActivityIndicator,
  RefreshControl,
} from 'react-native';
import React, {useEffect} from 'react';
import {useTheme} from 'react-native-paper';
import {useDispatch} from 'react-redux';

import {ITheme} from '~/theme/interfaces';
import ScreenWrapper from '~/beinComponents/ScreenWrapper';
import Header from '~/beinComponents/Header';
import groupsActions from '../../redux/actions';
import {useKeySelector} from '~/hooks/selector';
import groupsKeySelector from '../../redux/keySelector';
import Text from '~/beinComponents/Text';
import EmptyScreen from '~/beinFragments/EmptyScreen';
import Divider from '~/beinComponents/Divider';
import CommunityMemberRequest from './CommunityMemberRequest';
import ApproveDeclineAllRequests from './ApproveDeclineAllRequests';
import {useBaseHook} from '~/hooks';

const CommunityPendingMembers = () => {
  const theme = useTheme() as ITheme;
  const styles = themeStyles(theme);
  const dispatch = useDispatch();
  const {t} = useBaseHook();
  const {id: communityId} = useKeySelector(groupsKeySelector.communityDetail);
  const {total, loading, canLoadMore, ids} = useKeySelector(
    groupsKeySelector.communityMemberRequests,
  );

  useEffect(() => {
    onRefresh();

    return () => {
      getData(); // to update the total member requests again on press back
    };
  }, [communityId]);

  const getData = (isRefreshing?: boolean) => {
    dispatch(
      groupsActions.getCommunityMemberRequests({communityId, isRefreshing}),
    );
  };

  const onLoadMore = () => {
    getData();
  };

  const onRefresh = () => {
    getData(true);
  };

  const renderEmpty = () => {
    if (loading) return null;
    return (
      <EmptyScreen
        source={'addUsers'}
        title="groups:text_no_pending_members_notice"
        description="groups:text_pending_request_notice_community"
      />
    );
  };

  const renderListHeader = () => {
    if (loading || total === 0) return null;
    return (
      <View style={styles.requestHeader}>
        <Text.H5 testID="community_pending_members.request_title">{`${total} ${t(
          'common:text_request',
          {
            count: total,
          },
        )}`}</Text.H5>
      </View>
    );
  };

  const renderItem = ({item}: {item: number}) => {
    return <CommunityMemberRequest requestId={item} />;
  };

  const renderListFooter = () => {
    return (
      !loading &&
      canLoadMore &&
      ids.length > 0 && (
        <View
          style={styles.listFooter}
          testID="community_pending_members.loading_more_indicator">
          <ActivityIndicator />
        </View>
      )
    );
  };

  return (
    <ScreenWrapper testID="CommunityPendingMembers" isFullView>
      <Header title={t('settings:title_pending_members')} />

      <FlatList
        testID="flatlist"
        style={styles.flatList}
        data={ids}
        renderItem={renderItem}
        keyExtractor={(item, index) => `requests_${item}_${index}`}
        ListEmptyComponent={renderEmpty}
        ListHeaderComponent={renderListHeader}
        ListFooterComponent={renderListFooter}
        showsVerticalScrollIndicator={false}
        onEndReached={onLoadMore}
        onEndReachedThreshold={0.1}
        ItemSeparatorComponent={() => <Divider style={styles.divider} />}
        refreshControl={
          <RefreshControl
            refreshing={loading}
            onRefresh={onRefresh}
            tintColor={theme.colors.borderDisable}
          />
        }
      />

      {ids.length > 0 && <ApproveDeclineAllRequests />}
    </ScreenWrapper>
  );
};

const themeStyles = (theme: ITheme) => {
  const {spacing} = theme;

  return StyleSheet.create({
    flatList: {
      marginHorizontal: spacing.margin.large,
    },
    requestHeader: {
      marginVertical: spacing.margin.base,
    },
    divider: {
      marginBottom: spacing.margin.small,
    },
    listFooter: {
      height: 100,
      justifyContent: 'center',
      alignItems: 'center',
    },
  });
};

export default CommunityPendingMembers;
