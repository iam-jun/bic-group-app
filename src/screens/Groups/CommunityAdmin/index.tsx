import {StyleSheet, View} from 'react-native';
import React, {useEffect} from 'react';
import {ExtendedTheme, useTheme} from '@react-navigation/native';
import {useDispatch} from 'react-redux';

import ScreenWrapper from '~/beinComponents/ScreenWrapper';
import Header from '~/beinComponents/Header';
import Divider from '~/beinComponents/Divider';

import {useRootNavigation} from '~/hooks/navigation';
import {useKeySelector} from '~/hooks/selector';
import groupsKeySelector from '../redux/keySelector';
import Text from '~/beinComponents/Text';
import MenuItem from '~/beinComponents/list/items/MenuItem';
import modalActions from '~/store/modal/actions';
import groupStack from '~/router/navigator/MainStack/GroupStack/stack';
import groupsActions from '../redux/actions';
import spacing from '~/theme/spacing';
import {useMyPermissions} from '~/hooks/permissions';

const CommunityAdmin = () => {
  const theme: ExtendedTheme = useTheme();
  const styles = createStyles(theme);
  const dispatch = useDispatch();
  const {rootNavigation} = useRootNavigation();
  const {
    id: communityId,
    name,
    icon,
    can_manage_scheme,
    can_edit_info,
  } = useKeySelector(groupsKeySelector.communityDetail);
  const {total} = useKeySelector(groupsKeySelector.communityMemberRequests);

  const {hasPermissionsOnCurrentAudience, PERMISSION_KEY} = useMyPermissions();
  const canManageJoiningRequests = hasPermissionsOnCurrentAudience(
    'communities',
    communityId,
    PERMISSION_KEY.COMMUNITY.APPROVE_REJECT_JOINING_REQUESTS,
  );
  const canEditProfileInfo = hasPermissionsOnCurrentAudience(
    'communities',
    communityId,
    [
      PERMISSION_KEY.COMMUNITY.EDIT_INFORMATION,
      PERMISSION_KEY.COMMUNITY.EDIT_PRIVACY,
    ],
  );

  useEffect(() => {
    canManageJoiningRequests &&
      dispatch(groupsActions.getCommunityMemberRequests({communityId}));

    return () => {
      dispatch(groupsActions.resetCommunityMemberRequests());
    };
  }, [communityId]);

  const displayNewFeature = () => dispatch(modalActions.showAlertNewFeature());

  const onPressPendingMembers = () => {
    rootNavigation.navigate(groupStack.communityPendingMembers, {
      id: communityId,
    });
  };

  const onPressGeneralInfo = () => {
    rootNavigation.navigate(groupStack.generalInfo, {
      id: communityId,
      type: 'community',
    });
  };

  const onPressPermission = () => {
    rootNavigation.navigate(groupStack.communityPermission);
  };

  const onPressGroupStructure = () => {
    rootNavigation.navigate(groupStack.groupStructureSettings);
  };

  const renderModerating = () => (
    <>
      <Text.BodyM
        style={styles.headerTitle}
        color={theme.colors.neutral80}
        variant="bodyM"
        useI18n>
        settings:title_community_moderating
      </Text.BodyM>
      {!!canManageJoiningRequests && (
        <MenuItem
          testID={'community_admin.pending_members'}
          title={'settings:title_pending_members'}
          icon={'UserCheck'}
          iconProps={{
            icon: 'UserCheck',
            tintColor: theme.colors.purple50,
          }}
          notificationsBadgeNumber={total}
          notificationsBadgeProps={{maxNumber: 99, variant: 'alert'}}
          rightSubIcon="AngleRightSolid"
          onPress={onPressPendingMembers}
        />
      )}
      <MenuItem
        testID={'community_admin.pending_posts'}
        title={'settings:title_pending_posts'}
        icon={'FileExclamation'}
        iconProps={{
          icon: 'FileExclamation',
          tintColor: theme.colors.purple50,
        }}
        notificationsBadgeNumber={999}
        notificationsBadgeProps={{maxNumber: 99, variant: 'alert'}}
        rightSubIcon="AngleRightSolid"
        onPress={displayNewFeature}
      />
    </>
  );

  const renderSettings = () => (
    <>
      <Text.BodyM
        style={styles.headerTitle}
        color={theme.colors.neutral80}
        variant="bodyM"
        useI18n>
        settings:title_community_settings
      </Text.BodyM>
      {!!canEditProfileInfo && (
        <MenuItem
          testID="community_admin.profile_info"
          title="settings:title_profile_info"
          icon="Gear"
          iconProps={{icon: 'Gear', tintColor: theme.colors.purple50}}
          rightSubIcon="AngleRightSolid"
          onPress={onPressGeneralInfo}
        />
      )}
      {!!can_edit_info && ( //todo temp use can edit info, should use correct permission when BE update
        <MenuItem
          testID="community_admin.group_structure_settings"
          title="settings:title_group_structure"
          icon="CodeBranch"
          iconProps={{icon: 'CodeBranch', tintColor: theme.colors.purple50}}
          rightSubIcon="AngleRightSolid"
          onPress={onPressGroupStructure}
        />
      )}
      {!!can_manage_scheme && (
        <MenuItem
          testID="community_admin.permission_settings"
          title="settings:title_permission_settings"
          icon="FileLock"
          iconProps={{icon: 'FileLock', tintColor: theme.colors.purple50}}
          rightSubIcon="AngleRightSolid"
          onPress={onPressPermission}
        />
      )}
      <MenuItem
        testID="community_admin.post_settings"
        title="settings:title_post_settings"
        icon="Copy"
        iconProps={{icon: 'Copy', tintColor: theme.colors.purple50}}
        rightSubIcon="AngleRightSolid"
        onPress={displayNewFeature}
      />
    </>
  );

  return (
    <ScreenWrapper testID="CommunityAdmin" isFullView>
      <Header
        title={name}
        titleTextProps={{color: theme.colors.neutral80}}
        avatar={icon}
      />
      <Divider size={4} />
      <View style={styles.container}>
        {renderModerating()}
        {renderSettings()}
      </View>
    </ScreenWrapper>
  );
};

const createStyles = (theme: ExtendedTheme) => {
  return StyleSheet.create({
    container: {
      paddingTop: spacing.padding.large,
    },
    headerTitle: {
      marginHorizontal: spacing.margin.large,
      marginVertical: spacing.margin.base,
    },
  });
};

export default CommunityAdmin;
