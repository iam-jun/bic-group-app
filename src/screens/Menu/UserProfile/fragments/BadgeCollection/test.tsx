import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
