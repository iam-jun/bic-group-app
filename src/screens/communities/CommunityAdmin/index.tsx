import { StyleSheet, View } from 'react-native';
import React from 'react';
import { ExtendedTheme, useTheme } from '@react-navigation/native';

import ScreenWrapper from '~/beinComponents/ScreenWrapper';
import Header from '~/beinComponents/Header';
import Divider from '~/beinComponents/Divider';

import { useRootNavigation } from '~/hooks/navigation';
import Text from '~/baseComponents/Text';
import MenuItem from '~/beinComponents/list/items/MenuItem';
import groupStack from '~/router/navigator/MainStack/stacks/groupStack/stack';
import spacing from '~/theme/spacing';
import useCommunitiesStore, { ICommunitiesState } from '~/store/entities/communities';
import { PermissionKey } from '~/constants/permissionScheme';
import useMyPermissionsStore from '~/store/permissions';
import articleStack from '~/router/navigator/MainStack/stacks/articleStack/stack';

const CommunityAdmin = (props: any) => {
  const { params } = props.route;
  const communityId = params?.communityId;

  const theme: ExtendedTheme = useTheme();
  const styles = createStyles();
  const { rootNavigation } = useRootNavigation();
  const community = useCommunitiesStore((state: ICommunitiesState) => state.data[communityId]);
  const { name = '', groupId } = community || {};

  const { shouldHavePermission } = useMyPermissionsStore((state) => state.actions);
  const canEditProfileInfo = shouldHavePermission(
    groupId,
    [
      PermissionKey.EDIT_INFO,
      PermissionKey.EDIT_PRIVACY,
    ],
  );

  const onPressGeneralInfo = () => {
    rootNavigation.navigate(
      groupStack.generalInfo, {
        id: communityId,
        type: 'community',
      },
    );
  };

  const onPressScheduleContent = () => {
    rootNavigation.navigate(
      articleStack.articleScheduleContent, { groupId },
    );
  };

  const renderModerating = () => null;
  // return (
  //   <>
  //     <Text.BodyM
  //       style={styles.headerTitle}
  //       color={theme.colors.neutral80}
  //       variant="bodyM"
  //       useI18n
  //     >
  //       settings:title_community_moderating
  //     </Text.BodyM>
  //     <MenuItem
  //       testID="community_admin.pending_posts"
  //       title="settings:title_pending_posts"
  //       icon="FileExclamation"
  //       iconProps={{
  //         icon: 'FileExclamation',
  //         tintColor: theme.colors.purple50,
  //       }}
  //       notificationsBadgeNumber={999}
  //       notificationsBadgeProps={{ maxNumber: 99, variant: 'alert' }}
  //       rightSubIcon="AngleRightSolid"
  //       onPress={displayNewFeature}
  //     />
  //   </>
  // );

  const renderSettings = () => (
    <>
      <Text.BodyM
        style={styles.headerTitle}
        color={theme.colors.neutral80}
        variant="bodyM"
        useI18n
      >
        settings:title_community_settings
      </Text.BodyM>
      {!!canEditProfileInfo && (
        <MenuItem
          testID="community_admin.profile_info"
          title="settings:title_profile_info"
          icon="Gear"
          iconProps={{ icon: 'Gear', tintColor: theme.colors.purple50 }}
          rightSubIcon="AngleRightSolid"
          onPress={onPressGeneralInfo}
        />
      )}
      <MenuItem
        testID="community_admin.schedule_content"
        title="settings:title_schedule_content"
        icon="BallotCheck"
        iconProps={{ icon: 'BallotCheck', tintColor: theme.colors.purple50 }}
        rightSubIcon="AngleRightSolid"
        onPress={onPressScheduleContent}
      />
    </>
  );

  return (
    <ScreenWrapper testID="CommunityAdmin" isFullView>
      <Header
        title={name}
        titleTextProps={{ color: theme.colors.neutral80 }}
      />
      <Divider size={4} />
      <View style={styles.container}>
        {renderModerating()}
        {renderSettings()}
      </View>
    </ScreenWrapper>
  );
};

const createStyles = () => StyleSheet.create({
  container: {
    paddingTop: spacing.padding.large,
  },
  headerTitle: {
    marginHorizontal: spacing.margin.large,
    marginVertical: spacing.margin.base,
  },
});

export default CommunityAdmin;
