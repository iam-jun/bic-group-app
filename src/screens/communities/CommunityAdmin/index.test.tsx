/* eslint-disable @typescript-eslint/ban-ts-comment */
import React from 'react';
import { renderWithRedux } from '../../../test/testUtils';
import CommunityAdmin from './index';

describe('CommunityAdmin component', () => {
  it('should render data correctly', () => {
    const wrapper = renderWithRedux(<CommunityAdmin />);
    expect(wrapper).toMatchSnapshot();
  });
});
