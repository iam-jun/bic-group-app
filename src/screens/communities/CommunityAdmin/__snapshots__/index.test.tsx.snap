// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityAdmin component should render data correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
      },
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
  testID="CommunityAdmin"
>
  <View
    style={
      Object {
        "zIndex": 2,
      }
    }
    testID="header"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "#FFFFFF",
            "flexDirection": "row",
            "minHeight": 44,
            "paddingTop": 0,
            "paddingVertical": 4,
          },
          Object {
            "elevation": 2,
            "shadowColor": "rgba(0, 0, 0, 1)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.12,
            "shadowRadius": 1.5,
          },
          undefined,
        ]
      }
      testID="header.content"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "flex": 1,
            "flexDirection": "row",
            "overflow": "hidden",
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 20,
              "left": 20,
              "right": 20,
              "top": 20,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="header.back.button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "center",
                  "height": 44,
                  "justifyContent": "center",
                  "padding": 12,
                  "width": 44,
                },
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                testID="header.back"
              />
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "justifyContent": "center",
                "paddingVertical": 8,
              },
              Object {},
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "height": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#E3E6F2",
          "height": 4,
          "marginHorizontal": 0,
          "width": undefined,
        },
        Object {},
      ]
    }
    testID="divider"
  />
  <View
    style={
      Object {
        "paddingTop": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Light",
            "fontSize": 15,
          },
          Object {
            "color": "#181C32",
          },
          Object {
            "marginHorizontal": 16,
            "marginVertical": 12,
          },
        ]
      }
    >
      Community Moderating
    </Text>
    <View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="community_admin.pending_posts"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 16,
                "paddingVertical": 12,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                  Object {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "marginLeft": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#212745",
                    "fontFamily": "BeVietnamPro-Medium",
                    "fontSize": 15,
                  },
                  Object {},
                  undefined,
                ]
              }
              testID="menu_item.title"
            >
              Pending Posts
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#E90C22",
                  "borderRadius": 100,
                  "height": 20,
                  "justifyContent": "center",
                  "minWidth": 20,
                  "paddingBottom": 2,
                  "paddingHorizontal": 5,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#212745",
                    "fontFamily": "BeVietnamPro-Light",
                    "fontSize": 13,
                  },
                  Object {},
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "lineHeight": 16,
                      "textAlign": "center",
                    },
                    Object {},
                  ],
                ]
              }
            >
              99+
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 10,
                  "left": 10,
                  "right": 10,
                  "top": 10,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "marginLeft": 12,
                    },
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <View
                    testID="menu_item.right_sub_icon"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Light",
            "fontSize": 15,
          },
          Object {
            "color": "#181C32",
          },
          Object {
            "marginHorizontal": 16,
            "marginVertical": 12,
          },
        ]
      }
    >
      Community Settings
    </Text>
    <View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="community_admin.post_settings"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "paddingHorizontal": 16,
                "paddingVertical": 12,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            hitSlop={
              Object {
                "bottom": 10,
                "left": 10,
                "right": 10,
                "top": 10,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                  Object {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "marginLeft": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#212745",
                    "fontFamily": "BeVietnamPro-Medium",
                    "fontSize": 15,
                  },
                  Object {},
                  undefined,
                ]
              }
              testID="menu_item.title"
            >
              Post Settings
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 10,
                  "left": 10,
                  "right": 10,
                  "top": 10,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "marginLeft": 12,
                    },
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <View
                    testID="menu_item.right_sub_icon"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
